django/db/models/fields/related_descriptors.py

```diff
@@ -581,10 +582,17 @@ def _apply_rel_filters(self, queryset):
                 # that abuse create_reverse_many_to_one_manager() with reverse
                 # one-to-many relationships instead and break known related
                 # objects assignment.
-                rel_obj_id = tuple([
-                    getattr(self.instance, target_field.attname)
-                    for target_field in self.field.get_path_info()[-1].target_fields
-                ])
+                try:
+                    target_field = self.field.target_field
+                except FieldError:
+                    # The relationship has multiple target fields, use a tuple
+                    # for related object id.
+                    rel_obj_id = tuple(
```

Comment:
Doesn't `tuple([])` perform better?

Reply:
Supposedly but given this wasn't in a hot path I figured this form was more readable. I'll add back the list comprehension wrapping.